benches/nexmark/main.rs

```diff
@@ -1 +1,67 @@
+//! Nexmark benchmarks for DBSP
+//!
+//! Based on the queries defined at
+#![feature(is_some_with)]
+use anyhow::Result;
+use clap::Parser;
+use dbsp::{
+    circuit::Root,
+    nexmark::{
+        config::Config as NexmarkConfig, generator::config::Config as GeneratorConfig,
+        model::Event, queries::q1, NexmarkSource,
+    },
+    profile::CPUProfiler,
+    trace::{ord::OrdZSet, BatchReader},
+};
+use rand::prelude::ThreadRng;
 
+// TODO: Enable running specific tests
+fn main() -> Result<()> {
+    let nexmark_config = NexmarkConfig::parse();
+    let generator_config = GeneratorConfig::new(nexmark_config, 0, 0, 0, 0);
+
+    let cpu_profiler = CPUProfiler::new();
+    let root = Root::build(|circuit| {
+        // TODO(absoludity): CPUProfiler is not currently used in any of the
```

Comment:
The CPU profiler is useful for optimizing the performance, as it attributes CPU time to specific operators. I need to package it a bit better to make it easier to use, along with the memory profiler.  Both are fairly of rudimentary, but since we won't be building anything more serious in the immediate future, we should make it easy to consume the existing implementation.

But it is not meant for reporting the overall CPU/memory consumption of a benchmark.  We need some other tools for that.  I personally use `/usr/bin/time`, which gives you the three important numbers: memory, wall clock time, and CPU utilization. I haven't looked into what's the best way to do this in Rust.

Reply:
> We need some other tools for that. I personally use `/usr/bin/time`, which gives you the three important numbers: memory, wall clock time, and CPU utilization. I haven't looked into what's the best way to do this in Rust.

Yeah, I was using `time cargo bench --bench nexmark --features with-nexmark -- --first-event-rate=1000000` initially to get an idea (which includes any required compile too, not great). I'll look for a way to do so from within the harness.